src/Core/TileMesh.js

```diff
@@ -224,8 +219,10 @@ TileMesh.prototype.getCoordsForLayer = function getCoordsForLayer(layer) {
         } else {
             return OGCWebServiceHelper.computeTMSCoordinates(this, layer.extent, layer.origin);
         }
-    } else {
+    } else if (layer.extent.crs() == this.extent.crs()) {
         return [this.extent];
```

Comment:
It could but actually the `as()` function should be improved to support a 'no-clone' argument (currently we always clone the extent, even if the requested crs is the same).

Reply:
Done